src/mono/mono/component/hot_reload.c

```diff
@@ -1898,8 +1898,12 @@ apply_enclog_pass2 (Pass2Context *ctx, MonoImage *image_base, BaselineInfo *base
 		/* TODO: See CMiniMdRW::ApplyDelta for how to drive this.
 		 */
 		switch (func_code) {
-		case ENC_FUNC_DEFAULT: /* default */
+		case ENC_FUNC_DEFAULT: /* default */ {
+			if (!is_addition && token_table == MONO_TABLE_FIELD) {
+				add_member_typedef = log_token;
```

Comment:
Let's not do it like this.  the `add_member_XXX` variables are there because the ADD functions all come paired with another row with a DEFAULT function - it's kind of a hack to make sure we process them together.  we don't need it for rows that are self-contained.

Just move  the assert in the `MONO_TABLE_FIELD` case into the `is_addition` branch:

```
  if (is_addition) {
     g_assert (add_member_typedef);
    // the rest is all the same
  } else {
     // just use log_token instead of add_member_typedef
 }
```